presto-main/src/main/java/com/facebook/presto/execution/SharedBuffer.java

```diff
@@ -363,16 +326,7 @@ public synchronized void fail()
         }
 
         state.set(FAILED);
-
-        // clear the buffer
-        masterBuffer.clear();
-        bufferedBytes = 0;
-
-        // free queued page waiters
-        for (QueuedPage queuedPage : queuedPages) {
-            queuedPage.getFuture().set(null);
-        }
-        queuedPages.clear();
+        partitionToPageBuffer.values().forEach(pageBuffer -> pageBuffer.clear(false));
```

Comment:
the old code would free writers.  Is there a reason to not free them in the new code?  If not, then we don't need the parameter to clear.